### PR TITLE
redmine#6883: Implement body common control tls_min_version and body server control allowtlsversion.

### PR DESCRIPTION
**Pull request by jimis**

### CODE REVIEW

**Comment by jimis** [comment]
trigger build

**Comment by jimis** [comment]
acceptance tests will come in a separate pull request.

**Comment by kacfengine** [comment]
Both code and build look good!

File: libcfnet/tls_generic.c [hunk 1]
```diff
@@ -771,17 +775,71 @@ void TLSSetDefaultOptions(SSL_CTX *ssl_ctx)
...
+    const char *compiletime_min_version = "1.2";
+
+#ifndef  SSL_OP_NO_TLSv1_1
+#   define SSL_OP_NO_TLSv1_1 0x0                                /* nop */
+    compiletime_min_version = "1.1";
+
+# ifndef  SSL_OP_NO_TLSv1
+#   define SSL_OP_NO_TLSv1 0x0                                  /* nop */
+    compiletime_min_version = "1.0";
+# endif
```
Comment (kacfengine) [inline libcfnet/tls_generic.c]: Nice solution to the ifdef problem, btw.

**Pull request merged**